api/managementpb/dbaas/xtradb_cluster.proto

```diff
@@ -0,0 +1,217 @@
+
+// TODO move it to parent directory / package once API is v1-stable
+
+// TODO: find soluton to re-use:
+// import "github.com/percona-platform/dbaas-api/api/controller/common.proto";
+// import "github.com/percona-platform/dbaas-api/api/controller/xtra_db_cluster.proto";
+// with *.pb.go in difrent path (/gen/)
+
+// RunningOperation respresents a long-running operation.
+message RunningOperation {
+  // Progress from 0.0 to 1.0; can decrease compared to the previous value.
+  float progress = 1;
+  // Text describing the current operation progress step.
+  string message = 2;
+}
+
+// ComputeResources represents container computer resources requests or limits.
+message ComputeResources {
+  // CPUs in milliCPUs; 1000m = 1 vCPU.
+  int32 cpu_m = 1;
+  // Memory in bytes.
+  int64 memory_bytes = 2;
+}
+
+// XtraDBClusterState represents XtraDB cluster CR state.
+enum XtraDBClusterState {
+  // XTRA_DB_CLUSTER_STATE_INVALID represents unknown state.
+  XTRA_DB_CLUSTER_STATE_INVALID = 0;
+  // XTRA_DB_CLUSTER_STATE_CHANGING represents a cluster being changed (initializing).
+  XTRA_DB_CLUSTER_STATE_CHANGING = 1;
+  // XTRA_DB_CLUSTER_STATE_READY represents a cluster without pending changes (ready).
+  XTRA_DB_CLUSTER_STATE_READY = 2;
+  // XTRA_DB_CLUSTER_STATE_FAILED represents a failed cluster (error).
+  XTRA_DB_CLUSTER_STATE_FAILED = 3;
+  // XTRA_DB_CLUSTER_STATE_DELETING represents a cluster being deleting.
+  XTRA_DB_CLUSTER_STATE_DELETING = 4;
+}
+
+// XtraDBBackupState represents XtraDB backup CR state.
+enum XtraDBBackupState {
+  // XTRA_DB_BACKUP_STATE_INVALID represents unknown state.
+  XTRA_DB_BACKUP_STATE_INVALID = 0;
+  // XTRA_DB_BACKUP_STATE_RUNNING represents running backup (Starting, Running).
+  XTRA_DB_BACKUP_STATE_RUNNING = 1;
+  // XTRA_DB_BACKUP_STATE_SUCCEEDED represents succeeded backup (Succeeded).
+  XTRA_DB_BACKUP_STATE_SUCCEEDED = 2;
+  // XTRA_DB_BACKUP_STATE_FAILED represents failed backup (Failed).
+  XTRA_DB_BACKUP_STATE_FAILED = 3;
+}
+
+// XtraDBClusterParams represents XtraDB cluster parameters that can be updated.
+message XtraDBClusterParams {
+  // PXC container parameters.
+  message PXC {
+    // Requested compute resources.
+    ComputeResources compute_resources = 1 [
+      (validator.field) = {
+        msg_exists: true
+      }
+    ];
+  }
+  // ProxySQL container parameters.
+  message ProxySQL {
+    // Requested compute resources.
+    ComputeResources compute_resources = 1 [
+      (validator.field) = {
+        msg_exists: true
+      }
+    ];
+  }
+  // Cluster size.
+  int32 cluster_size = 1 [
+    (validator.field) = {
+      int_gt: 0
+    }
+  ];
+  // PXC container parameters.
+  PXC pxc = 2 [
+    (validator.field) = {
+      msg_exists: true
+    }
+  ];
+  // ProxySQL container parameters.
+  ProxySQL proxysql = 3 [
+    (validator.field) = {
+      msg_exists: true
+    }
+  ];
+}
+
+// XtraDBCluster service provides public methods for managing XtraDB clusters.
```

Comment:
Usually we put service in the end of the file.

Reply:
Moved.